lib/rolodex/route.ex

```diff
@@ -338,11 +338,33 @@ defmodule Rolodex.Route do
           verb: atom()
         }
 
+  @doc """
+  Checks to see if the given route matches any filter(s) stored in `Rolodex.Config`.
+  """
+  @spec matches_filter?(t(), Rolodex.Config.t()) :: boolean()
+  def matches_filter?(route, config)
+
+  def matches_filter?(route, %Config{filters: filters}) when is_list(filters) do
+    Enum.any?(filters, fn
+      filter_opts when is_map(filter_opts) ->
+        keys = Map.keys(filter_opts)
```

Comment:
I think you can use `match?` for this bit instead of taking keys and doing an equality check

Reply:
I ... can't get this to work? Maybe I just don't fully understand the pin operator, but pinning an entire map isn't working as I'd expect it to.

Examples:

```elixir
bar = "bar"
baz = "baz"
data = %{foo: "bar", bar: "baz"}

iex> match?(%{foo: ^bar}, data)
true

iex> to_match = %{foo: bar}
iex> match?(^to_match, data)
false
```

Reply:
Ah wait, I'm realizing that when you pin an entire map, the RHS needs to match the entire pinned map exactly.

Reply:
So we could do

```elixir
match?(^filter_opts, Map.take(route, Map.keys(filter_opts)))
```

But that feels like basically the same thing? Am I missing something obvious here?